### DISCUSSION TITLE
Where can I find the source of tess core and tess core_vanilla

### DISCUSSION THREAD

**Discussion by rmast** [scribeocr/scribeocr#36] [Q&A] [answered]
I find two versions of a wasm translation of Tesseract in this directory:
https://github.com/scribeocr/scribeocr/tree/master/tess

Are these the results of compiling https://github.com/scribeocr/tesseract.js and https://github.com/scribeocr/tesseract.js-core?
Which one is which?

**Comment by rmast** [comment]
Do these tesseract versions contain custom code, or are they just recompiled updates of patches on Tesseract itself?
Is there a way to just focus on the custom differences at the most abstract source level?

**Comment by rmast** [comment]
I guess I found some references to original sources of these wasms:
https://github.com/scribeocr/scribeocr/blob/master/package.json: 
  "homepage": "https://github.com/scribeocr/scribeocr#readme",
  "dependencies": {
    "canvas": "^2.11.2",
    "commander": "^11.1.0",
    "tesseract.js": "scribeocr/tesseract.js#**39c4e79**",
    "web-worker": "~1.2.0"
  }
  
  https://github.com/scribeocr/tesseract.js/blob/7f1cb3242d917a6090ee9db9cd615267dfdd019e/package-lock.json:
  
  {
  "name": "tesseract.js",
  "version": "5.0.4",
  "lockfileVersion": 3,
  "requires": true,
  "packages": {
    "": {
      "name": "tesseract.js",
      "version": "5.0.4",
      "hasInstallScript": true,
      "license": "Apache-2.0",
      "dependencies": {
        "bmp-js": "^0.1.0",
        "idb-keyval": "^6.2.0",
        "is-electron": "^2.2.2",
        "is-url": "^1.2.4",
        "node-fetch": "^2.6.9",
        "opencollective-postinstall": "^2.0.3",
        "regenerator-runtime": "^0.13.3",
        "tesseract.js-core": "scribeocr/tesseract.js-core#**ed2b922**",
        "wasm-feature-detect": "^1.2.11",
        "zlibjs": "^0.3.1"
      },
  
  but still no trace of which of the source trees (vanilla or the other, are both in use?) is used when selecting these:
  
![image](https://github.com/scribeocr/scribeocr/assets/3341558/8becbe8b-b1b9-4ee6-abd1-99893ca4681b)

**Comment by rmast** [comment]
![image](https://github.com/scribeocr/scribeocr/assets/3341558/77f8f471-9000-40f0-a925-980be150f371)
![image](https://github.com/scribeocr/scribeocr/assets/3341558/d4f7a032-44f9-4c32-9090-e4a1c666b0c7)

**Comment by rmast** [comment]
https://github.com/scribeocr/scribeocr/commit/812199a4b77f2a4ec3bd547023d270077df78fa6
![image](https://github.com/scribeocr/scribeocr/assets/3341558/31a897f5-46e9-4f8a-99e9-f597b8cbf9d2)

**Comment by Balearica** [comment, marked as answer]
Scribe OCR uses a modified version of Tesseract.js, which is documented in the repo [here](https://github.com/naptha/tesseract.js).  In the "vanilla" build, the version of Tesseract used is nearly identical to the version used for Tesseract.js, which itself is nearly identical to the main Tesseract project.  All changes are minor, and facilitate running Tesseract in the browser, rather than making changes to the recognition model.

In the build modified for Scribe OCR, more substantial changes to Tesseract are made.  There is no definitive list of changes, however most of the changes have to do with font detection, text/noise detection, and layout analysis.

The repos that both versions are compiled from can be found in the Scribe OCR fork of Tesseract.js-core, which is [here](https://github.com/scribeocr/tesseract.js-core/tree/master).  These would be in the [third_party](https://github.com/scribeocr/tesseract.js-core/tree/master/third_party) and [third_party_vanilla](https://github.com/scribeocr/tesseract.js-core/tree/master/third_party_vanilla) directories.
